sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2355,6 +2359,28 @@ class Dataset[T] private[sql](
   }
 
   /**
+   * Collect a Dataset to an ArrowRecordBatch.
+   *
+   * @group action
+   * @since 2.2.0
+   */
+  @DeveloperApi
+  def collectAsArrow(
+      allocator: RootAllocator = new RootAllocator(Long.MaxValue)): ArrowRecordBatch = {
```

Comment:
The encapsulated return value (particularly when/if we are able to perform the `Array[InternalRow]` conversion on the task executors, should ideally be a collection of row batches. But the internal structure of the return value should preferably be hidden from interface here. A sort of `ArrowPayload` data structure that we can put on the wire later